docs/kubernetes/cross-cluster.md

```diff
@@ -0,0 +1,177 @@
+# Accessing filestore instance from workloads in multiple clusters
+
+>**Pre-requisites:** Filestore driver must be installed on all the clusters. Follow the driver install steps [here](https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/README.md#kubernetes-development)
+
+>**Pre-requisites:** Ensure that the clusters are created in the same [VPC](https://cloud.google.com/vpc/docs/vpc) as the filestore instance
+
+>**Attention:** Care must be taken in deleting the PersistentVolume resources, as there may be workloads in different clusters which may still point to the given filestore instance. Failure to do so, may result in pods stuck in a Terminating state.
+
+The following example demonstrates the usage of a dynamically created filestore instance by workloads deployed in two clusters in a single gcloud project (say `ClusterA` and `ClusterB`). The same example can be extended to multiple clusters. One of the cluster (here `ClusterA`) would dynamically create the filestore instance and map to the pods of a deployment in that cluster. Other clusters will re-use the filestore instance in a pre-provisioned PersitentVolume and use it in workloads.
+
+1. Create `StorageClass` in all the clusters.
+
+    If you haven't created a `StorageClass` yet, create one first:
+
+    ```console
+    $ kubectl apply -f ./examples/kubernetes/cross-cluster/sc.yaml
+    ```
+
+    If a non-default network is used for the filestore instance, provide a network paramter to the storage class.
+
+    ```yaml
+    apiVersion: storage.k8s.io/v1
+    kind: StorageClass
+    metadata:
+    name: csi-filestore
+    provisioner: filestore.csi.storage.gke.io
+    parameters:
```

Comment:
Maybe set a retain reclaim policy here? People will be copying this code, and for multi-cluster access probably retain is the safe policy to start with unless you really know what you're doing.

Reply:
Good point. Updated the doc with retain policy